nativelib/src/main/scala/scala/scalanative/regex/MachineInput.scala

```diff
@@ -192,7 +192,9 @@ object MachineInput {
       hayStack match {
         case hayStack: String =>
           hayStack.indexOf(needle, pos)
-        case hayStack: StringBuilder =>
+        case hayStack: java.lang.StringBuilder =>
+          hayStack.indexOf(needle, pos)
+        case hayStack: collection.mutable.StringBuilder =>
```

Comment:
You want both here?

Reply:
Yes, I don't want to change previous semantics, but rather adapt it to using java.lang.StringBuilder by default